labelbox/client.py

```diff
@@ -1597,3 +1596,49 @@ def get_model_slice(self, slice_id) -> ModelSlice:
         """
         res = self.execute(query_str, {"id": slice_id})
         return Entity.ModelSlice(self, res["getSavedQuery"])
+
+    class DeleteFeatureFromOntologyResult:
+        archived: bool
+        deleted: bool
+
+    def delete_feature_schema_from_ontology(
+            self, ontology_id: str,
+            feature_schema_id: str) -> DeleteFeatureFromOntologyResult:
+        """
+        Deletes or archives a feature schema from an ontology.
+        If the feature schema is a root level node with associated labels, it will be archived.
+        If the feature schema is a nested node in the ontology and does not have associated labels, it will be deleted.
+        If the feature schema is a nested node in the ontology and has associated labels, it will not be deleted.
+
+        Args:
+            ontology_id (str): The ID of the ontology.
+            feature_schema_id (str): The ID of the feature schema.
+
+        Returns:
+            DeleteFeatureFromOntologyResult: The result of the feature schema removal.
+
+        Example:
+            >>> client.remove_feature_schema_from_ontology(<ontology_id>, <feature_schema_id>)
+        """
+        ontology_endpoint = self.rest_endpoint + "/ontologies/" + urllib.parse.quote(
+            ontology_id) + "/feature-schemas/" + urllib.parse.quote(
+                feature_schema_id)
+        response = requests.delete(
+            ontology_endpoint,
+            headers=self.headers,
+        )
+
+        if response.status_code == requests.codes.ok:
+            response_json = response.json()
+            if response_json['archived'] == True:
+                logger.info(
+                    'Feature schema was archived from the ontology because it had associated labels.'
+                )
+            elif response_json['deleted'] == True:
+                logger.info(
+                    'Feature schema was successfully removed from the ontology')
+            return response_json
```

Comment:
Can we actually create an instance of `DeleteFeatureFromOntologyResult` as the return object, rather than returning `response_json` directly? This will ensure we have consistent typing and prevent API changes from breaking customer scripts, if there's additional fields added to the response json.

Reply:
[Updated](https://github.com/Labelbox/labelbox-python/pull/958/commits/2afab0f4d5920c2021fc2454f84405777b2337df).